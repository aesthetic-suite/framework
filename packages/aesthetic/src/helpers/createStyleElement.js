/**
 * @copyright   2017, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 * @flow
 */

export default function createStyleElement(id: string): HTMLElement {
  const style = document.createElement('style');
  style.type = 'text/css';
  style.id = `aesthetic-${id}`;

  // $FlowIgnore We know the document exists
  document.head.appendChild(style);

  return style;
}
