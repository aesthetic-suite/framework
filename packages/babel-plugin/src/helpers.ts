import { types as t } from '@babel/core';
import { createDebugger } from '@boost/debug';

export const debug = createDebugger('aesthetic:babel');

export function isFunction(node: t.Node) {
  return t.isFunctionExpression(node) || t.isArrowFunctionExpression(node);
}
