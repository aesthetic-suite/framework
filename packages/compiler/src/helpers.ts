import { ColorShade } from '@aesthetic/system';

export function formatShade(value: number | string): ColorShade {
	return String(value).padStart(2, '0').slice(0, 2) as ColorShade;
}

export function formatUnit(value: number): string {
	return value.toFixed(2).replace('.00', '');
}
