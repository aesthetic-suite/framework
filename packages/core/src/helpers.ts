import { Theme } from '@aesthetic/system';
import { objectLoop } from '@aesthetic/utils';
import { SheetParams } from './types';

export function createCacheKey(params: SheetParams, type: string): string {
  let key = type;

  // Since all other values are scalars, we can just join the values.
  // This is 3x faster than JSON.stringify(), and 1.5x faster than Object.values()!
  objectLoop(params, (value) => {
    key += value;
  });

  return key;
}

export function createDefaultParams(theme: Theme<{}>, params: SheetParams): Required<SheetParams> {
  return {
    contrast: theme.contrast,
    deterministic: false,
    direction: 'ltr',
    scheme: theme.scheme,
    theme: theme.name,
    unit: 'px',
    vendor: false,
    ...params,
  };
}
