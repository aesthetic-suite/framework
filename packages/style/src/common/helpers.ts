/* eslint-disable no-magic-numbers */

import { Value, MaybeValue, CSS, Sheet } from '@aesthetic/types';
import { hyphenate } from '@aesthetic/utils';
import { IMPORT_RULE, STYLE_RULE } from './constants';

export function insertRule(sheet: Sheet, rule: CSS, index?: number): number {
  try {
    return sheet.insertRule(rule, index ?? sheet.cssRules.length);
  } catch {
    // Vendor prefixed properties, pseudos, etc, that are inserted
    // into different vendors will trigger a failure. For example,
    // `-moz` or `-ms` being inserted into WebKit.
    // There's no easy way around this, so let's just ignore the
    // error so that subsequent styles are inserted.
    // istanbul ignore next
    return -1;
  }
}

export function insertAtRule(sheet: Sheet, rule: CSS): number {
  const { length } = sheet.cssRules;
  let index = 0;

  // At-rules must be inserted before normal style rules.
  for (let i = 0; i <= length; i += 1) {
    index = i;

    if (sheet.cssRules[i]?.type === STYLE_RULE) {
      break;
    }
  }

  return insertRule(sheet, rule, index);
}

export function insertImportRule(sheet: Sheet, rule: CSS): number {
  const { length } = sheet.cssRules;
  let index = 0;

  // Import rules must be inserted at the top of the style sheet,
  // but we also want to persist the existing order.
  for (let i = 0; i <= length; i += 1) {
    index = i;

    if (sheet.cssRules[i]?.type !== IMPORT_RULE) {
      break;
    }
  }

  return insertRule(sheet, rule, index);
}

export function isAtRule(value: string): boolean {
  return value[0] === '@';
}

export function isImportRule(value: string): boolean {
  return value.slice(0, 7) === '@import';
}

export function isMediaRule(value: string): boolean {
  return value.slice(0, 6) === '@media';
}

export function isNestedSelector(value: string): boolean {
  const char = value[0];

  return (
    char === ':' ||
    char === '[' ||
    char === '>' ||
    char === '~' ||
    char === '+' ||
    char === '*' ||
    char === '|'
  );
}

export function isSupportsRule(value: string): boolean {
  return value.slice(0, 9) === '@supports';
}

const unitlessProperties = new Set<string>();

[
  'animationIterationCount',
  'borderImage',
  'borderImageOutset',
  'borderImageSlice',
  'borderImageWidth',
  'columnCount',
  'columns',
  'flex',
  'flexGrow',
  'flexPositive',
  'flexShrink',
  'flexNegative',
  'flexOrder',
  'fontWeight',
  'gridArea',
  'gridRow',
  'gridRowEnd',
  'gridRowSpan',
  'gridRowStart',
  'gridColumn',
  'gridColumnEnd',
  'gridColumnSpan',
  'gridColumnStart',
  'lineClamp',
  'lineHeight',
  'maskBorder',
  'maskBorderOutset',
  'maskBorderSlice',
  'maskBorderWidth',
  'opacity',
  'order',
  'orphans',
  'tabSize',
  'widows',
  'zIndex',
  'zoom',
  // SVG
  'fillOpacity',
  'floodOpacity',
  'stopOpacity',
  'strokeDasharray',
  'strokeDashoffset',
  'strokeMiterlimit',
  'strokeOpacity',
  'strokeWidth',
].forEach((property) => {
  unitlessProperties.add(property);
  unitlessProperties.add(hyphenate(property));
});

export function isUnitlessProperty(property: string): boolean {
  return unitlessProperties.has(property);
}

export function isValidValue(property: string, value: MaybeValue | boolean): value is Value {
  if (value === null || value === undefined || value === true || value === false || value === '') {
    if (__DEV__) {
      // eslint-disable-next-line no-console
      console.warn(`Invalid value "${value}" for "${property}".`);
    }

    return false;
  }

  return true;
}

export function isVariable(value: string): boolean {
  return value.slice(0, 2) === '--';
}

export function joinQueries(prev: string | undefined, next: string): string {
  if (prev) {
    return `${prev} and ${next}`;
  }

  return next;
}
