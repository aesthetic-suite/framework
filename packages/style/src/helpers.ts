/* eslint-disable no-magic-numbers */

import { Value, MaybeValue } from '@aesthetic/types';
import { hyphenate } from '@aesthetic/utils';

export function isAtRule(value: string): boolean {
  return value[0] === '@';
}

export function isImportRule(value: string): boolean {
  return value.slice(0, 7) === '@import' && value[7] === ' ';
}

export function isNestedSelector(value: string): boolean {
  const char = value[0];

  return (
    char === ':' ||
    char === '[' ||
    char === '>' ||
    char === '~' ||
    char === '+' ||
    char === '*' ||
    char === '|'
  );
}

const unitlessProperties = new Set<string>();

[
  'animationIterationCount',
  'borderImage',
  'borderImageOutset',
  'borderImageSlice',
  'borderImageWidth',
  'columnCount',
  'columns',
  'flex',
  'flexGrow',
  'flexPositive',
  'flexShrink',
  'flexNegative',
  'flexOrder',
  'fontWeight',
  'gridArea',
  'gridRow',
  'gridRowEnd',
  'gridRowSpan',
  'gridRowStart',
  'gridColumn',
  'gridColumnEnd',
  'gridColumnSpan',
  'gridColumnStart',
  'lineClamp',
  'lineHeight',
  'maskBorder',
  'maskBorderOutset',
  'maskBorderSlice',
  'maskBorderWidth',
  'opacity',
  'order',
  'orphans',
  'tabSize',
  'widows',
  'zIndex',
  'zoom',
  // SVG
  'fillOpacity',
  'floodOpacity',
  'stopOpacity',
  'strokeDasharray',
  'strokeDashoffset',
  'strokeMiterlimit',
  'strokeOpacity',
  'strokeWidth',
].forEach((property) => {
  unitlessProperties.add(property);
  unitlessProperties.add(hyphenate(property));
});

export function isUnitlessProperty(property: string): boolean {
  return unitlessProperties.has(property);
}

export function isValidValue(property: string, value: MaybeValue | boolean): value is Value {
  if (value === null || value === undefined || value === true || value === false || value === '') {
    if (__DEV__) {
      console.warn(`Invalid value "${value}" for "${property}".`);
    }

    return false;
  }

  return true;
}

export function isVariable(value: string): boolean {
  return value.slice(0, 2) === '--';
}
