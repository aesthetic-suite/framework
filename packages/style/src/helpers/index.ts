import createAtomicCacheKey from './createAtomicCacheKey';
import formatConditions from './formatConditions';
import formatDeclaration from './formatDeclaration';
import formatDeclarationBlock from './formatDeclarationBlock';
import formatRule from './formatRule';
import formatVariable from './formatVariable';
import formatVariableName from './formatVariableName';
import getDocumentStyleSheet from './getDocumentStyleSheet';
import isAtRule from './isAtRule';
import isFontFaceFule from './isFontFaceFule';
import isImportRule from './isImportRule';
import isInvalidValue from './isInvalidValue';
import isKeyframesRule from './isKeyframesRule';
import isMediaRule from './isMediaRule';
import isNestedSelector from './isNestedSelector';
import isSupportsRule from './isSupportsRule';
import isUnitlessProperty from './isUnitlessProperty';
import isVariable from './isVariable';

export {
  createAtomicCacheKey,
  formatConditions,
  formatDeclaration,
  formatDeclarationBlock,
  formatRule,
  formatVariable,
  formatVariableName,
  getDocumentStyleSheet,
  isAtRule,
  isFontFaceFule,
  isImportRule,
  isInvalidValue,
  isKeyframesRule,
  isMediaRule,
  isNestedSelector,
  isSupportsRule,
  isUnitlessProperty,
  isVariable,
};
