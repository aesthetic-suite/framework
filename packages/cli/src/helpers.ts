import { CONFIG_FOLDER } from '@aesthetic/compiler';
import { Path } from '@boost/common';
import { CWD } from './constants';

export function getConfigFolderDir(name: string): Path {
  return new Path(CWD, CONFIG_FOLDER, name);
}

export function validateSystemName(value: string) {
  if (value && !value.match(/^[-a-z0-9]+$/giu)) {
    throw new Error('Name may only contain numbers, letters, and dashes.');
  }
}
