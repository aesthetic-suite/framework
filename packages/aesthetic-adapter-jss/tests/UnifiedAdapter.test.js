/* eslint-disable sort-keys */

import { create } from 'jss';
import preset from 'jss-preset-default';
import UnifiedJSSAdapter from '../src/UnifiedAdapter';
import {
  FONT_ROBOTO_FLAT_SRC,
  FONT_CIRCULAR_MULTIPLE_FLAT_SRC,
  KEYFRAME_FADE,
  SYNTAX_UNIFIED_FULL,
  SYNTAX_ATTRIBUTE,
  SYNTAX_CHARSET,
  SYNTAX_DESCENDANT,
  SYNTAX_FALLBACKS,
  SYNTAX_FONT_FACE,
  SYNTAX_FONT_FACE_MIXED,
  SYNTAX_FONT_FACE_MULTIPLE,
  SYNTAX_GLOBAL,
  SYNTAX_IMPORT,
  SYNTAX_IMPORT_MULTIPLE,
  SYNTAX_KEYFRAMES,
  SYNTAX_MEDIA_QUERY,
  SYNTAX_MULTI_SELECTOR,
  SYNTAX_NAMESPACE,
  SYNTAX_PAGE,
  SYNTAX_PROPERTIES,
  SYNTAX_PSEUDO,
  SYNTAX_SUPPORTS,
  SYNTAX_VIEWPORT,
} from '../../../tests/mocks';
import { renderJSSStyles } from '../../../tests/helpers';

describe('aesthetic-adapter-jss/UnifiedAdapter', () => {
  let instance;

  beforeEach(() => {
    const jss = create();
    jss.setup(preset());

    instance = new UnifiedJSSAdapter(jss);
  });

  it('transforms style declarations into class names', () => {
    expect(instance.transform(instance.create(SYNTAX_UNIFIED_FULL).button)).toBe('button-0-1-1');
  });

  it('converts unified syntax to native syntax', () => {
    expect(instance.syntax.convert(SYNTAX_UNIFIED_FULL)).toEqual({
      '@font-face': [FONT_ROBOTO_FLAT_SRC],
      '@keyframes fade': KEYFRAME_FADE,
      button: {
        margin: 0,
        padding: '6px 12px',
        border: '1px solid #2e6da4',
        borderRadius: 4,
        display: 'inline-block',
        cursor: 'pointer',
        fontFamily: 'Roboto',
        fontWeight: 'normal',
        lineHeight: 'normal',
        whiteSpace: 'nowrap',
        textDecoration: 'none',
        textAlign: 'center',
        backgroundColor: '#337ab7',
        verticalAlign: 'middle',
        color: 'rgba(0, 0, 0, 0)',
        animationName: 'fade',
        animationDuration: '.3s',
        '&:hover': {
          backgroundColor: '#286090',
          borderColor: '#204d74',
        },
        '&::before': {
          content: '"â˜…"',
          display: 'inline-block',
          verticalAlign: 'middle',
          marginRight: 5,
        },
        '@media (max-width: 600px)': {
          padding: '4px 8px',
        },
      },
    });
  });

  it('handles globals', () => {
    expect(instance.syntax.convert(SYNTAX_GLOBAL)).toEqual({
      '@global': {
        body: { margin: 0 },
        html: { height: '100%' },
        a: {
          color: 'red',
          '&:hover': {
            color: 'darkred',
          },
        },
      },
    });

    instance.transform(instance.create(SYNTAX_GLOBAL));

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles properties', () => {
    expect(instance.syntax.convert(SYNTAX_PROPERTIES)).toEqual(SYNTAX_PROPERTIES);

    expect(instance.transform(instance.create(SYNTAX_PROPERTIES).props)).toBe('props-0-4-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles pseudo selectors', () => {
    expect(instance.syntax.convert(SYNTAX_PSEUDO)).toEqual({
      pseudo: {
        position: 'fixed',
        '&:hover': {
          position: 'static',
        },
        '&::before': {
          position: 'absolute',
        },
      },
    });

    expect(instance.transform(instance.create(SYNTAX_PSEUDO).pseudo)).toBe('pseudo-0-5-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @charset', () => {
    expect(instance.syntax.convert(SYNTAX_CHARSET)).toEqual({
      '@charset': 'utf8',
    });

    expect(instance.transform(instance.create(SYNTAX_CHARSET))).toBe('');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @fallbacks', () => {
    expect(instance.syntax.convert(SYNTAX_FALLBACKS)).toEqual({
      fallback: {
        background: 'linear-gradient(...)',
        display: 'flex',
        fallbacks: [{ background: 'red' }, { display: 'box' }, { display: 'flex-box' }],
      },
    });

    expect(instance.transform(instance.create(SYNTAX_FALLBACKS).fallback)).toBe('fallback-0-7-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @font-face', () => {
    expect(instance.syntax.convert(SYNTAX_FONT_FACE)).toEqual({
      '@font-face': [FONT_ROBOTO_FLAT_SRC],
      font: {
        fontFamily: 'Roboto',
        fontSize: 20,
      },
    });

    instance.syntax.fontFaces = {};

    expect(instance.transform(instance.create(SYNTAX_FONT_FACE).font)).toBe('font-0-8-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @import', () => {
    expect(instance.syntax.convert(SYNTAX_IMPORT)).toEqual({
      '@import': './some/path.css',
    });

    expect(instance.transform(instance.create(SYNTAX_IMPORT))).toBe('');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @keyframes', () => {
    expect(instance.syntax.convert(SYNTAX_KEYFRAMES)).toEqual({
      '@keyframes fade': KEYFRAME_FADE,
      animation: {
        animationName: 'fade',
        animationDuration: '3s, 1200ms',
        animationIterationCount: 'infinite',
      },
    });

    instance.syntax.keyframes = {};

    expect(instance.transform(instance.create(SYNTAX_KEYFRAMES).animation)).toBe(
      'animation-0-10-1',
    );

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @media', () => {
    expect(instance.syntax.convert(SYNTAX_MEDIA_QUERY)).toEqual({
      media: {
        color: 'red',
        '@media (max-width: 1000px)': {
          color: 'green',
        },
        '@media (min-width: 300px)': {
          color: 'blue',
        },
      },
    });

    expect(instance.transform(instance.create(SYNTAX_MEDIA_QUERY).media)).toBe('media-0-11-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @namespace', () => {
    expect(instance.syntax.convert(SYNTAX_NAMESPACE)).toEqual({
      '@namespace': 'url(http://www.w3.org/1999/xhtml)',
    });

    expect(instance.transform(instance.create(SYNTAX_NAMESPACE))).toBe('');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @page', () => {
    expect(() => {
      instance.transform(instance.create(SYNTAX_PAGE));
    }).toThrowError();
  });

  it('handles @supports', () => {
    expect(instance.syntax.convert(SYNTAX_SUPPORTS)).toEqual({
      sup: {
        display: 'block',
        '@supports (display: flex)': {
          display: 'flex',
        },
        '@supports not (display: flex)': {
          float: 'left',
        },
      },
    });

    expect(instance.transform(instance.create(SYNTAX_SUPPORTS).sup)).toBe('sup-0-14-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles @viewport', () => {
    expect(instance.syntax.convert(SYNTAX_VIEWPORT)).toEqual({
      '@viewport': {
        width: 'device-width',
        orientation: 'landscape',
      },
    });

    expect(instance.transform(instance.create(SYNTAX_VIEWPORT))).toBe('');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles descendant selectors', () => {
    expect(instance.syntax.convert(SYNTAX_DESCENDANT)).toEqual({
      list: {
        margin: 0,
        padding: 0,
        '&> li': {
          listStyle: 'bullet',
        },
      },
    });

    expect(instance.transform(instance.create(SYNTAX_DESCENDANT).list)).toBe('list-0-16-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles attribute selectors', () => {
    expect(instance.syntax.convert(SYNTAX_ATTRIBUTE)).toEqual({
      attr: {
        display: 'block',
        '&[disabled]': {
          opacity: 0.5,
        },
      },
    });

    expect(instance.transform(instance.create(SYNTAX_ATTRIBUTE).attr)).toBe('attr-0-17-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles multiple selectors (comma separated)', () => {
    expect(instance.syntax.convert(SYNTAX_MULTI_SELECTOR)).toEqual({
      multi: {
        cursor: 'pointer',
        '&:disabled': { cursor: 'default' },
        '&[disabled]': { cursor: 'default' },
        '&> span': { cursor: 'default' },
      },
    });

    expect(instance.transform(instance.create(SYNTAX_MULTI_SELECTOR).multi)).toBe('multi-0-18-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles multiple @font-face', () => {
    expect(instance.syntax.convert(SYNTAX_FONT_FACE_MULTIPLE)).toEqual({
      '@font-face': FONT_CIRCULAR_MULTIPLE_FLAT_SRC,
      font: {
        fontFamily: 'Circular',
        fontSize: 20,
      },
    });

    instance.syntax.fontFaces = {};

    expect(instance.transform(instance.create(SYNTAX_FONT_FACE_MULTIPLE).font)).toBe('font-0-19-1');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles mixed @font-face', () => {
    expect(instance.syntax.convert(SYNTAX_FONT_FACE_MIXED)).toEqual({
      '@font-face': [FONT_ROBOTO_FLAT_SRC, ...FONT_CIRCULAR_MULTIPLE_FLAT_SRC],
    });

    instance.syntax.fontFaces = {};

    instance.create(SYNTAX_FONT_FACE_MIXED);

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });

  it('handles multiple @import', () => {
    expect(instance.syntax.convert(SYNTAX_IMPORT_MULTIPLE)).toEqual({
      '@import': ['./some/path.css', './another/path.css'],
    });

    expect(instance.transform(instance.create(SYNTAX_IMPORT_MULTIPLE))).toBe('');

    expect(renderJSSStyles(instance)).toMatchSnapshot();
  });
});
