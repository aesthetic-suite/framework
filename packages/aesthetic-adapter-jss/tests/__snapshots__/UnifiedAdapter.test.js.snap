// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @charset 1`] = `"@charset utf8;"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @fallbacks 1`] = `".fallback-0-7-1 {background: red;display: box;display: flex-box;display: flex;background: linear-gradient(...);}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @font-face 1`] = `"@font-face {font-family: Roboto;font-style: normal;font-weight: normal;src: local('Robo'), url('fonts/Roboto.woff2') format('woff2'), url('fonts/Roboto.ttf') format('truetype');}.font-0-8-1 {font-size: 20px;font-family: Roboto;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @import 1`] = `"@import ./some/path.css;"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @keyframes 1`] = `"@keyframes fade {from {opacity: 0;}to {opacity: 1;}}.animation-0-10-1 {animation-name: fade;animation-duration: 3s, 1200ms;animation-iteration-count: infinite;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @media 1`] = `".media-0-11-1 {color: red;}@media (min-width: 300px) {.media-0-11-1 {color: blue;}}@media (max-width: 1000px) {.media-0-11-1 {color: green;}}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @namespace 1`] = `"@namespace url(http://www.w3.org/1999/xhtml);"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @supports 1`] = `".sup-0-14-1 {display: block;}@supports (display: flex) {.sup-0-14-1 {display: flex;}}@supports not (display: flex) {.sup-0-14-1 {float: left;}}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles @viewport 1`] = `"@viewport {width: device-width;orientation: landscape;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles attribute selectors 1`] = `".attr-0-17-1 {display: block;}.attr-0-17-1[disabled] {opacity: 0.5;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles descendant selectors 1`] = `".list-0-16-1 {margin: 0;padding: 0;}.list-0-16-1> li {list-style: bullet;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles globals 1`] = `"body {margin: 0;}html {height: 100%;}a {color: red;}a:hover {color: darkred;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles mixed @font-face 1`] = `"@font-face {font-family: Roboto;font-style: normal;font-weight: normal;src: local('Robo'), url('fonts/Roboto.woff2') format('woff2'), url('fonts/Roboto.ttf') format('truetype');}@font-face {font-family: Circular;font-style: normal;font-weight: normal;src: url('fonts/Circular.woff2') format('woff2');}@font-face {font-family: Circular;font-style: italic;font-weight: normal;src: url('fonts/Circular-Italic.woff2') format('woff2');}@font-face {font-family: Circular;font-style: normal;font-weight: 300;src: url('fonts/Circular-Light.woff2') format('woff2');}@font-face {font-family: Circular;font-style: normal;font-weight: 700;src: url('fonts/Circular-Bold.woff2') format('woff2');}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles multiple @font-face 1`] = `"@font-face {font-family: Circular;font-style: normal;font-weight: normal;src: url('fonts/Circular.woff2') format('woff2');}@font-face {font-family: Circular;font-style: italic;font-weight: normal;src: url('fonts/Circular-Italic.woff2') format('woff2');}@font-face {font-family: Circular;font-style: normal;font-weight: 300;src: url('fonts/Circular-Light.woff2') format('woff2');}@font-face {font-family: Circular;font-style: normal;font-weight: 700;src: url('fonts/Circular-Bold.woff2') format('woff2');}.font-0-19-1 {font-size: 20px;font-family: Circular;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles multiple @import 1`] = `"@import ./some/path.css;@import ./another/path.css;"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles multiple selectors (comma separated) 1`] = `".multi-0-18-1 {cursor: pointer;}.multi-0-18-1:disabled {cursor: default;}.multi-0-18-1[disabled] {cursor: default;}.multi-0-18-1> span {cursor: default;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles properties 1`] = `".props-0-4-1 {color: black;margin: 10px;display: inline;}"`;

exports[`aesthetic-adapter-jss/UnifiedAdapter handles pseudo selectors 1`] = `".pseudo-0-5-1 {position: fixed;}.pseudo-0-5-1:hover {position: static;}.pseudo-0-5-1::before {position: absolute;}"`;
